### PR TITLE
feat: build first version of extension 🥳

### PR DESCRIPTION
**Pull request by T3kstiil3**
GitHub issue/pull request detail

## Description

Init first version of the extension app
A simpler iframe of the web app for the moment
BTW the only solution to try it :

https://developer.chrome.com/extensions/getstarted
or 

## Screenshot

image

## TODO

- [x] Works on Firefox 👎

### CODE REVIEW

**Comment by T3kstiil3** [comment]
Waiting Google approve to test private beta version !

image

**Comment by T3kstiil3** [comment]
https://github.com/mdn/webextensions-examples/blob/master/menu-demo/manifest.json

**Comment by T3kstiil3** [comment]
https://developer.mozilla.org/fr/docs/Mozilla/Add-ons/WebExtensions/Your_first_WebExtension
https://developer.chrome.com/extensions/getstarted
https://github.com/fav-sh/extension/blob/master/src/manifest.json

File: extension/package.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "extension",
+  "version": "1.0.0",
+  "description": "",
+  "scripts": {
+    "build": "echo \"Soon a script to build extension and upload it 🥳\" && exit 1"
```
Comment (ytvnr) [inline extension/package.json]: 👍

File: extension/package.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "extension",
+  "version": "1.0.0",
```
Comment (ytvnr) [inline extension/package.json]:
```suggestion
  "version": "alpha",
```

File: front/.env.local [hunk 3]
```diff
@@ -1 +1,2 @@
 VUE_APP_EXTENSION_BASE_URL=http://localhost:1337/
+VUE_APP_MODE=d
```
Comment (ytvnr) [inline front/.env.local]: what is it ?

File: front/src/firebase/init.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ const firebaseConfig = require('./config.json');
 
 // Initialize Firebase
 firebase.initializeApp(firebaseConfig);
-firebase.auth().setPersistence(firebase.auth.Auth.Persistence.SESSION);
+firebase.auth().setPersistence(firebase.auth.Auth.Persistence.LOCAL);
```
Comment (ytvnr) [inline front/src/firebase/init.js]: is there impact for "web" mode vs extension ?

File: front/src/views/About.vue [hunk 5]
```diff
@@ -1,16 +1,24 @@
 <template>
-  <div class="about">
-    #about
-  </div>
+    <div class="about">
+        <ul>
+            <li>version: {{ mode }}.{{ version }}</li>
```
Comment (ytvnr) [inline front/src/views/About.vue]: Maybe we can id too to the footer ?

File: extension/package.json (hunk 2 above)
Comment (T3kstiil3) [reply extension/package.json]: https://semver.org/spec/v1.0.0-beta.html should include a version too
Should i switch to 1.0.0-alpha

File: front/src/firebase/init.js (hunk 4 above)
Comment (T3kstiil3) [reply front/src/firebase/init.js]: We need to have LOCAL for web like for extension

File: front/.env.local (hunk 3 above)
Comment (T3kstiil3) [reply front/.env.local]: image
Its useful to know if extension if dev or prod version

File: front/src/views/About.vue (hunk 5 above)
Comment (T3kstiil3) [reply front/src/views/About.vue]: Fur sure in a next version ?

**Pull request merged**